awserr.go

```diff
@@ -42,3 +36,50 @@ func IsAWSErrExtended(err error, code string, message string, origErrMessage str
 
 	return false
 }
+
+// IsAWSErrCode returns true if the error matches all these conditions:
+//  * err is of type awserr.Error
+//  * Error.Code() equals code
+func IsAWSErrCode(err error, code string) bool {
```

Comment:
We did some workshopping on the functions, since some of the names like `IsAWSErrCodeMessageContains()` didn't read smoothly.

We came up with moving them to a package `tfawserr` and the following renaming:
* `IsAWSErrCode()` to `ErrCodeEquals()`
* `IsAWSErrCodeContains()` to `ErrCodeContains()`
* `IsAWSErrCodeMessageContains()` to `ErrMessageContains()`
* `IsAWSErrRequestFailureStatusCode()` to `ErrStatusCodeEquals()`

What do you think?

Reply:
Makes sense.
Do we still want to keep `IsAWSErr` in the top-level package? Removing it would be a breaking change.

Reply:
We should probably leave it for backwards compatibility, even though we're pre-1.0 :)

Reply:
I feel comfortable removing `IsAWSErr` as the only usage I can find is in the AWS Provider.